crates/re_arrow_store/src/store_write.rs

```diff
@@ -185,11 +185,30 @@ impl DataStore {
         }
 
         // This is valuable information even for a timeless timepoint!
-        self.metadata_registry.insert(*row_id, timepoint.clone());
+        match self.metadata_registry.entry(*row_id) {
+            std::collections::btree_map::Entry::Vacant(entry) => {
+                entry.insert(timepoint.clone());
+            }
+            // NOTE: When saving and loading data from disk, it's very possible that we try to
+            // insert data for a single `RowId` in multiple calls!
+            std::collections::btree_map::Entry::Occupied(mut entry) => {
+                let entry = entry.get_mut();
+                for (&timeline, &time) in timepoint {
+                    entry.insert(timeline, time);
```

Comment:
It shouldn't be possible for different columns of the same `RowId` to be inserted in multiple calls (that would be different log calls and so different `RowId`s).

This comment refers to the case where different timelines of the same `RowId` are being inserted in multiple calls, which currently happens anytime buckets are saved to and loaded from disk, since the same `RowId` will likely appear in different buckets (because buckets are per-timeline).
I.e. `assert!(!entry.insert(timeline, time));` should never fail here.

Of course in practice if end users use the low-level APIs to forcibly re-use `RowId`s across log calls then all bets are off but nothing we can do about that besides printing a warning and moving on.